### DISCUSSION TITLE
Is there a way to set the visible area of a Plot at runtime?

### DISCUSSION THREAD

**Discussion by CatThingy** [emilk/egui#1328] [Q&A] [answered]
I want the width and height of my Plot to be within some variable bounds so that all the data remains visible. I tried using variables in `include_x` and `include_y` but the bounds of the Plot didn't update when the variables did, except when they reached 0.

Edit: Having a way to lock the Plot into the double-clicked mode would also be an option.

**Comment by benedikt-jensen** [comment]
Exactly what I need as well. It is already possible to get the plot bounds, which is great, but it only works one way.  It would be nice to have the option to set the bounds in the same way.

**Comment by lvaroqui** [comment]
Hi @emilk,

I stumbled into this issue as well, I would like to be able to control precisely from app code the bounds of my plots. 

I though maybe we could split the `Plot` API in two parts.

To maintain backward compatibility `Plot` would not change its API at all. We could create a new struct `BasicPlot` that would contain `Plot` logic without auto_bounds / drag / zoom and simply have a `bounds` setter used to set the bounds of the plot. This `BasicPlot` could be exposed to the public API to let people finely control the bounds while the `Plot` struct will use a BasicPlot internally and automatically fill the bound like it does right now.

Do you think that would be a good idea?

I'm willing to draft PR!

Maybe this is too complex for nothing and simply adding the bounds method would be enough.

**Reply by avsaase** [reply]
Personally I would happily accept a breaking to have this feature added.

**Comment by flxzt** [comment]
I also think that being able to set the viewport of the plot would be very useful!

**Comment by lvaroqui** [comment, marked as answer]
There is now the `set_plot_bounds` which does the job perfectly.

**Comment by benedikt-jensen** [comment]
Great, thanks for letting us know!

**Reply by lvaroqui** [reply]
Note that there was a little bug with it that got fix a few days ago and that should be able in the next release!

**Reply by lvaroqui** [reply]
See https://github.com/emilk/egui/issues/2652
